addons/mail/static/src/components/chat_window/chat_window.js

```diff
@@ -129,9 +129,11 @@ class ChatWindow extends Component {
      * @private
      */
     _saveThreadScrollTop() {
-        this.chatWindow.update({
-            threadInitialScrollTop: this._threadRef.comp.getScrollTop(),
-        });
+        if (this.chatWindow.threadViewer) {
```

Comment:
This condition necessary? If you only call it when there is a `_threadRef.comp`, surely there would be a threadViewer?

Is that called in a destroy context where record was already deleted?

Reply:
It was defensive coding, maybe too much defensive

Reply:
It is necessary as in `onClickedHeader` if the chat window is the "new message" chat window, then there is no threadViewer and there is no need to store scroll position

Reply:
(as we removed threadComp condition from onClickedHeader)

Reply:
See other comments.